src/profile.c

```diff
@@ -125,6 +129,18 @@ void Profile_Print(RedisModule_Reply *reply, void *ctx) {
         if (profile_verbose)
           RedisModule_ReplyKV_Double(reply, "Pipeline creation time",
             (double)(req->pipelineBuildTime / CLOCKS_PER_MILLISEC));
+      //Print total GIL time
+        if (profile_verbose){
+          if (RunInThread()){
+            RedisModule_ReplyKV_Double(reply, "Total GIL time", req->qiter.GILTime);
+          } else {
+            struct timespec rpEndTime;
+            clock_gettime(CLOCK_MONOTONIC, &rpEndTime);
+            RedisModule_ReplyKV_Double(reply, "Total GIL time",
+              (double)(rpEndTime.tv_nsec - req->qiter.initTime.tv_nsec)/1000000.0);
```

Comment:
Isn't `tv_nsec` holding only up to a second, and are the whole seconds counted in the `tv_sec`?
I think we have to use a full diff function here. Also, what happens if the start time is "XX:XX:n.99" and the end is "XX:XX:(n+1).01"? Is the diff negative?

Reply:
See `rs_timersub`